arviz/stats/stats.py

```diff
@@ -1287,14 +1295,28 @@ def summary(
                 )
                 extra_metric_names.append(stat_func.__name__)
 
+    metrics: List[xr.Dataset] = []
+    metric_names: List[str] = []
     if extend and kind in ["all", "stats"]:
-        mean = dataset.mean(dim=("chain", "draw"), skipna=skipna)
-
-        sd = dataset.std(dim=("chain", "draw"), ddof=1, skipna=skipna)
-
-        hdi_post = hdi(dataset, hdi_prob=hdi_prob, multimodal=False, skipna=skipna)
-        hdi_lower = hdi_post.sel(hdi="lower", drop=True)
-        hdi_higher = hdi_post.sel(hdi="higher", drop=True)
+        if stat_focus == "mean":
+            mean = dataset.mean(dim=("chain", "draw"), skipna=skipna)
+
+            sd = dataset.std(dim=("chain", "draw"), ddof=1, skipna=skipna)
+
+            hdi_post = hdi(dataset, hdi_prob=hdi_prob, multimodal=False, skipna=skipna)
+            hdi_lower = hdi_post.sel(hdi="lower", drop=True)
+            hdi_higher = hdi_post.sel(hdi="higher", drop=True)
+            metrics.extend((mean, sd, hdi_lower, hdi_higher))
+            metric_names.extend(("mean", "sd", f"hdi_{100 * alpha / 2:g}%", f"hdi_{100 * (1 - alpha / 2):g}%"))
+        elif stat_focus == "median":
+            median = dataset.median(dim=("chain", "draw"), skipna=skipna)
+
+            mad = stats.median_abs_deviation(dataset, dims=("chain", "draw"))
+            eti_post = dataset.quantile((alpha/2, 1-alpha/2), dim=("chain", "draw"), skipna=skipna)
+            eti_lower = eti_post.isel(quantile=0, drop=True)
+            eti_higher = eti_post.isel(quantile=1, drop=True)
+            metrics.extend((median, mad, eti_lower, eti_higher))
+            metric_names.extend(("median", "mad", f"eti_{100 * alpha / 2:g}%", f"eti_{100 * (1 - alpha / 2):g}%"))
 
     if circ_var_names:
```

Comment:
Now circmean and circstd can use the ones from xarray_einstats directly instead of coding the wrappers here.